### DISCUSSION TITLE
[Feature Request] provide an API for adding buttons to the top right corner of the Textarea

### DISCUSSION THREAD

**Discussion by Jackeriss** [heroui-inc/heroui#4738] [Feature requests]
### Is your feature request related to a problem? Please describe.

I want to implement the effect like the NextUI doc: show a copy button when hover the textarea.
<img width="744" alt="image" src="https://github.com/user-attachments/assets/07f045fe-2a47-4d73-ba93-a0c0274931b0">

### Describe the solution you'd like

It will be great if NextUI can provide an API for adding buttons to the top right corner of the Textarea.

### Describe alternatives you've considered

or maybe Textarea can support copy button original.

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1379/[feature-request]-provide-an-api-for-adding-buttons-to-the-top-right">ENG-1379 [Feature Request] provide an API for adding buttons to the top right corner of the Textarea</a></p>
